payjoin/src/send/error.rs

```diff
@@ -29,6 +44,8 @@ pub(crate) enum InternalBuildSenderError {
     ChangeIndexPointsAtPayee,
     InputWeight(crate::psbt::InputWeightError),
     AddressType(crate::psbt::AddressTypeError),
+    #[cfg(feature = "v2")]
+    FailedToPersistSender(ErrorBox),
```

Comment:
The feature flag is a smell that implies that there might be a better home for this variant. Perhaps the suggestion that separates concerns and fixes save/load asymmetry can be applied to the send side as well.

Reply:
This didn't end up being neccecary after fixing the asymmetry of save/load. `SenderBuilderError` is left untouched. The multiparty sender does use the `v2::SenderBuilder` and errors. There we use a noop persister and expect(...) the errors. I think thats a fine trade off. Alternatively we can refactor the multiparty sender a bit to have its own error type for SenderBuilder but I'd like to revisit that when we add persistence for the mulitpary mod